server/canvas/api.py

```diff
@@ -12,14 +12,22 @@ def canvas_api_request(canvas_course, method, endpoint, **kwargs):
     response.raise_for_status()
     return response.json()
 
-def score_endpoint(canvas_assignment, enrollment):
-    # We can use the student ID directly here. See
-    # https://bcourses.berkeley.edu/doc/api/file.object_ids.html
-    return '/courses/{}/assignments/{}/submissions/sis_user_id:{}'.format(
-        canvas_assignment.canvas_course.external_id,
-        canvas_assignment.external_id,
-        enrollment.sid,
-    )
+def canvas_api_get_list(canvas_course, endpoint, **kwargs):
+    url = 'https://' + canvas_course.api_domain + '/api/v1' + endpoint
+    headers = {
+        'Accept': 'application/json',
+        'Authorization': 'Bearer ' + canvas_course.access_token,
+    }
+    params = kwargs.pop('params', {})
+    params['per_page'] = 100
+    response = requests.get(url, headers=headers, params=params, **kwargs)
+    response.raise_for_status()
+    results = response.json()
+    while 'next' in response.links:
+        response = requests.get(response.links['next']['url'], headers=headers)
+        response.raise_for_status()
+        results.extend(response.json())
```

Comment:
We can cache getting all students for a bit (since every upload has to do this).